src/integrationtests/java/com/aws/greengrass/integrationtests/ConfigTest.java

```diff
@@ -78,6 +87,90 @@ void GIVEN_Greengrass_with_mqtt_bridge_WHEN_multiple_config_changes_consecutivel
         assertEquals(1, numRestarts.get());
     }
 
+    @TestWithMqtt5Broker
+    @WithKernel("mqtt5_connect_options.yaml")
+    void GIVEN_Greengrass_with_mqtt_bridge_WHEN_connect_options_set_in_config_THEN_local_client_uses_configured_values(Broker broker, ExtensionContext context)
+            throws Exception {
+        long expectedSessionExpiryInterval = 10;
+        long expectedMaximumPacketSize = 100;
+        int expectedReceiveMaximum = 1000;
+
+        // verify that the config is correctly read
+        assertEquals(expectedSessionExpiryInterval, testContext.getConfig().getSessionExpiryInterval());
+        assertEquals(expectedMaximumPacketSize, testContext.getConfig().getMaximumPacketSize());
+        assertEquals(expectedReceiveMaximum, testContext.getConfig().getReceiveMaximum());
+
+        // verify that the brokerClient config values are correctly set in the local client
+        assertEquals(expectedSessionExpiryInterval, testContext.getLocalV5Client().getSessionExpiryInterval());
+        assertEquals(expectedMaximumPacketSize, testContext.getLocalV5Client().getMaximumPacketSize());
+        assertEquals(expectedReceiveMaximum, testContext.getLocalV5Client().getReceiveMaximum());
+
+        Topics config = testContext.getKernel().locate(MQTTBridge.SERVICE_NAME).getConfig()
+                .lookupTopics(CONFIGURATION_CONFIG_KEY).lookupTopics("brokerClient");
+
+        // publish a small message and verify that it is received
+        String topic = "topic/toLocal";
+        Set<String> topics = new HashSet<>();
+        topics.add(topic);
+
+        MqttMessage expectedMessage = MqttMessage.builder()
+                .topic(topic)
+                .payload("abc".getBytes(StandardCharsets.UTF_8))
+                .userProperties(Collections.singletonList(new UserProperty("key", "val")))
+                .responseTopic("response topic")
+                .messageExpiryIntervalSeconds(1234L)
+                .payloadFormat(Publish.PayloadFormatIndicator.UTF8)
+                .contentType("contentType")
+                .build();
+
+        Consumer<MqttMessage> messageHandler = (message) -> {
+          assertEquals(Arrays.toString(expectedMessage.getPayload()), Arrays.toString(message.getPayload()));
+        };
+
+        testContext.getLocalV5Client().updateSubscriptions(topics, messageHandler);
+        testContext.getLocalV5Client().publish(
+                MqttMessage.builder()
+                        .topic(topic)
+                        .payload("abc".getBytes(StandardCharsets.UTF_8))
+                        .userProperties(Collections.singletonList(new UserProperty("key", "val")))
+                        .responseTopic("response topic")
+                        .messageExpiryIntervalSeconds(1234L)
+                        .payloadFormat(Publish.PayloadFormatIndicator.UTF8)
+                        .contentType("contentType")
+                        .build());
+        TimeUnit.MILLISECONDS.sleep(100);
```

Comment:
is there a way we can get rid of the sleep?

Reply:
Looking back I think I could try using a `Pair<CompletableFuture<Void>, Consumer<MqttMessage>` and wait on the completable future instead of sleeping to ensure the handler is triggered